packages/commerce-sdk-react/src/hooks/useAuthHelper.ts

```diff
@@ -40,13 +51,32 @@ export function useAuthHelper<Mutation extends AuthHelper>(
 > {
     const auth = useAuthContext()
     if (!auth[mutation]) throw new Error(`Unknown login helper mutation: ${mutation}`)
-
+    const queryClient = useQueryClient()
     // I'm not sure if there's a way to avoid this type assertion, but, I'm fairly confident that
     // it is safe to do, as it seems to be simply re-asserting what we already know.
     type Method = Auth[Mutation]
     type PromisedData = ReturnType<Method>
     type Data = PromisedData extends Promise<infer D> ? D : never
     type Variables = [] extends Parameters<Method> ? void : Parameters<Method>[0]
     const method = auth[mutation].bind(auth) as MutationFunction<Data, Variables>
-    return useMutation(auth.whenReady(method))
+    return useMutation(auth.whenReady(method), {
+        onSuccess(data, options) {
+            const getCacheUpdates = cacheUpdateMatrix[mutation]
+
+            if (getCacheUpdates) {
+                const cacheUpdates = getCacheUpdates(options, data)
+                updateCache(queryClient, cacheUpdates, data)
+            }
+        }
+    })
 }
+
+const cacheUpdateMatrix: CacheUpdateMatrix = {
+    loginRegisteredUserB2C: noop,
+    loginGuestUser: noop,
```

Comment:
Got it. That makes sense. I think because the helpers are very similar to the api hook implementations it makes sense to make the same choices for these. If we decided to flop the other way, we can do it for all of them.